### PR TITLE
MGMT-15240: Add Unit test job for OpenShift Appliance

### PR DESCRIPTION
**Pull request by nmagnezi**

### CODE REVIEW

File: ci-operator/config/openshift/appliance/openshift-appliance-master.yaml [hunk 1]
```diff
@@ -44,6 +44,11 @@ tests:
     make lint
   container:
     from: test-bin
+- as: unit-test
+  commands: |
+    make unit-test
```
[the remark this answers is not in the document]
Comment (nmagnezi) [reply ci-operator/config/openshift/appliance/openshift-appliance-master.yaml:49]: We should do this once the secret for the appliance exists. 
The pod will likely fail if we mount a non-existing secret.

File: ci-operator/config/openshift/appliance/openshift-appliance-master.yaml [hunk 2]
```diff
@@ -44,6 +44,11 @@ tests:
...
+  container:
+    from: test-bin
```
[the remark this answers is not in the document]
Comment (nmagnezi) [reply ci-operator/config/openshift/appliance/openshift-appliance-master.yaml:52]: Good point. Done.

File: ci-operator/config/openshift/appliance/openshift-appliance-master.yaml [hunk 3]
```diff
@@ -44,6 +44,12 @@ tests:
...
+  run_if_changed: ^(cmd/.*|data/.*|hack/.*|pkg/.*|Dockerfile\..*|Makefile|go\.mod|go\.sum)$
```
Comment (danielerez) [inline ci-operator/config/openshift/appliance/openshift-appliance-master.yaml:52]: Maybe remove 'data' and 'hack', there are less relevant.

**Pull request merged**